packages/fxa-settings/src/pages/Signin/SigninTokenCode/index.tsx

```diff
@@ -58,10 +63,18 @@ const SigninTokenCode = ({
     // }
   };
 
-  const onSubmit = () => {
+  const onSubmit = async () => {
     try {
+      GleanMetrics.loginConfirmation.submit();
       // Check confirmation code
+
       // Log success event
+      // The await of isDone is not entirely necessary when we are not
+      // redirecting the user to an RP.  However at the time of implementation
+      // for the Glean ping the redirect logic has not been implemented.
+      GleanMetrics.loginConfirmation.success();
+      await GleanMetrics.isDone();
```

Comment:
Should this always be awaited before navigating away from the app and to an RP? Asking just to understand best practice.

Reply:
Yes, just so we don't unload the page (therefore stop code execution) before the page can send all the pings.